include/xgboost/generic_parameters.h

```diff
@@ -40,7 +40,7 @@ struct GenericParameter : public dmlc::Parameter<GenericParameter> {
         .describe("The primary GPU device ordinal.");
     DMLC_DECLARE_FIELD(n_gpus)
         .set_default(0)
-        .set_lower_bound(-1)
+        .set_lower_bound(0)
```

Comment:
i'm hoping no one uses the moving target (the head version) in their testing, as this is a change in behavior in their configuration.

Reply:
Yes this is a breaking change if you are using multi-gpu.